src/components/Daos/DaosPage.tsx

```diff
@@ -56,15 +59,34 @@ class DaosPage extends React.Component<IProps, IState> {
     });
   }
 
-  onSearchChange = (e: any) => {
+  onSearchChange = async (e: any) => {
     this.setState({ search: e.target.value });
+
+    // If search string greater than 2 search on server for any other DAOs not yet loaded that match this search
+    if (e.target.value.length > 2) {
+      const arc = getArc();
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      const foundDaos = await arc.daos({ orderBy: "name", orderDirection: "asc", where: { name_contains: e.target.value } }, { fetchAllData: true }).pipe(first()).toPromise();
+      this.setState({ searchDaos: foundDaos });
+    } else {
+      this.setState({ searchDaos: [] });
+    }
   }
 
   public render(): RenderOutput {
     const { currentAccountProfile, data, fetchMore } = this.props;
-    const search = this.state.search.toLowerCase();
+    const search = this.state.search.length > 2 ? this.state.search.toLowerCase() : "";
```

Comment:
Why do we need `state.search` when we have `state.searchDAOs`?  Does the input control really need to get its value from the state?

We don't appear to be using the local `search` variable at all.

Reply:
the search var is used to actually filter the DAOs in the UI, yes it needs to be tracked in the state

Reply:
OK, I see now that it is being used, couldn't see that before, maybe because the lines are long.  But the check for the length of the array is redundant.